src/Planners/include/BipedalLocomotion/Planners/Contact.h

```diff
@@ -59,6 +59,26 @@ struct Contact
      * Type of contact.
      */
     ContactType type {ContactType::FULL};
+
+    /**
+     * @brief The equality operator.
+     *
+     * @param other The other object used for the comparison.
+     * @return True if the contacts are the same, false otherwise.
+     */
+    bool operator==(const Contact& other) const
+    {
+        auto doubleEq = [](const double d1, const double d2) {
+            return std::abs(d1 - d2) < std::numeric_limits<double>::epsilon();
```

Comment:
Mhh this seems to be counter-intuitive. For all floating point units, == is clearly defined as "the floating point representation is exactly the same", and that is quite known. Redefining an operator== that instead follows a different criteria may not be obvious. Why you need to define such a custom operator?

Reply:
I'm just used to use this approach for comparing floating point numbers. Using `==` on floating point representation nowadays raise warnings on modern compilers. I think that a machine epsilon of difference is tolerable, isn't it?

Reply:
> I think that a machine epsilon of difference is tolerable, isn't it?

In C++ `operator==` is a synonym for having a real comparison, so I think that adding such an operator will violate the so-called [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) . If I am not wrong, that is true also in Python (see https://docs.python.org/3/whatsnew/3.5.html#pep-485-a-function-for-testing-approximate-equality that instead defines a isclose method for that). I would personally not trade a specific developer convenient for API clarity.

Reply:
I don't have anything against comparing the raw floating-point representation, what bothers me is mainly the `-Wfloat-equal` warning. What do @GiulioRomualdi and @S-Dafarra think? Currently I'm using the comparison operator [only in the test](https://github.com/diegoferigo/bipedal-locomotion-framework/blob/0f06adbbd29828e2f1d5d30e236a613c3a0f122a/bindings/python/tests/test_contact.py#L62) that I think would succeed also with a raw comparison.

Reply:
Ack, I never experienced the `-Wfloat-equal` warning, and from what I understand it is not part of `-Wall`, and that make sense as there are cases in which comparing exactly with a floating point make sense, as for example in these tests. I checked also on Eigen and they do  similar choice, there is both the `operator==` that does the exact comparison (see https://eigen.tuxfamily.org/dox/classEigen_1_1MatrixBase.html#title67), and the isApprox method that instead takes in input (optionally) a threshold: https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c .

Reply:
I removed the tolerance from the comparison in a5c7aad, as expected the test keep passing.